k9mail/src/test/java/com/fsck/k9/mailstore/AttachmentResolverTest.java

```diff
@@ -0,0 +1,105 @@
+package com.fsck.k9.mailstore;
+
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import android.net.Uri;
+
+import com.fsck.k9.mail.BodyPart;
+import com.fsck.k9.mail.Multipart;
+import com.fsck.k9.mail.Part;
+import com.fsck.k9.message.extractors.AttachmentInfoExtractor;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@SuppressWarnings("unchecked")
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = 21)
+public class AttachmentResolverTest {
+    public static final Uri ATTACHMENT_TEST_URI_1 = Uri.parse("uri://test/1");
+    public static final Uri ATTACHMENT_TEST_URI_2 = Uri.parse("uri://test/2");
+
+
+    @Test
+    public void buildCidMap__onPartWithNoBody__shouldReturnEmptyMap() throws Exception {
+        AttachmentInfoExtractor attachmentInfoExtractor = mock(AttachmentInfoExtractor.class);
+        Part part = mock(Part.class);
+        when(part.getContentId()).thenReturn(null);
+
+        Map<String,Uri> result = AttachmentResolver.buildCidToAttachmentUriMap(attachmentInfoExtractor, part);
+
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void buildCidMap__onMultipartWithNoParts__shouldReturnEmptyMap() throws Exception {
+        AttachmentInfoExtractor attachmentInfoExtractor = mock(AttachmentInfoExtractor.class);
+        Part multipartPart = mock(Part.class);
+        Multipart multipartBody = mock(Multipart.class);
+        when(multipartPart.getBody()).thenReturn(multipartBody);
```

Comment:
`Part` and friends are simple enough that we should use the real thing for testing rather than mocking them. That should make tests more robust against changes of the implementation details of the class under test.